### ISSUE TITLE
ECONNREFUSED 127.0.0.1:80 when following Oauth tutorial

### ISSUE THREAD

**Issue by kinostl** [feathersjs/feathers#1564]
### Steps to reproduce

- Use `feathers generate app` as normal with the oauth options
- Add a "discord" section to the oauth configuration and do other appropriate configurations
- run application as normal, and use the /oauth/discord workflow on the default configured 3030 port
- receive `error: connect ECONNREFUSED 127.0.0.1:80 `

Works as expected if port is changed to 80 and project is run as root.

### Expected behavior
Access redirected page with access_token in header when following tutorial.

### Actual behavior
Receive error page with following error when redirect is turned off.

```
error: connect ECONNREFUSED 127.0.0.1:80 
```
```javascript
{
  "errno": "ECONNREFUSED",
  "code": 500,
  "syscall": "connect",
  "address": "127.0.0.1",
  "port": 80,
  "hook": {
    "type": "before",
    "method": "create",
    "path": "authentication",
    "params": {
      "authStrategies": [
        "discord"
      ],
      "authentication": null
    },
    "data": {
      "strategy": "discord",
      "access_token": "token",
      "refresh_token": "token",
      "raw": {
        "access_token": "token",
        "scope": "identify",
        "token_type": "Bearer",
        "expires_in": 604800,
        "refresh_token": "token"
      }
    }
  }
}
```

### System configuration

**NodeJS version**:
node v12.4.0
npm 6.11.3

**Operating System**:
Ubuntu 19.04

**Browser Version**:
Firefox Quantum 69.0

**Comment by aljaxus** [comment]
Can confirm this behaviour

**Comment by daffl** [comment]
We need the authentication configuration otherwise nobody will be able to reproduce. It may even be fixed by [configuring the host](https://github.com/simov/grant#configuration) with the proper address.

**Comment by aljaxus** [comment]
> We need the authentication configuration otherwise nobody will be able to reproduce. It may even be fixed by [configuring the host](https://github.com/simov/grant#configuration) with the proper address.

Yeah, sorry for the delay. I was on a bus and couldn't provide more info at that time.
This is my "authentication" section of the config. Domain `lvh.me` points to `127.0.0.1`. As we can see, this is a repeating pattern - both of us, me and @kinostl, have localhost set as our host string. Well mine is a fqdn, but as said it resolves to 127.0.0.1

Though this issue does not exist when I use GitHub or Google as oAuth providers, meaning that Discord might be doing something like a source domain check on their side.

```
"authentication": {
    "entity": "user",
    "service": "users",
    "secret": "SECRET",
    "authStrategies": [
      "jwt",
      "local"
    ],
    "jwtOptions": {
      "header": {
        "typ": "access"
      },
      "audience": "http://lvh.me",
      "issuer": "feathers",
      "algorithm": "HS256",
      "expiresIn": "1d"
    },
    "local": {
      "usernameField": "email",
      "passwordField": "password"
    },
    "oauth": {
      "redirect": "/",
      "defaults": {
        "host": "lvh.me",
        "state": true
      },
      "google": {
        "key": "IDTOKEN",
        "secret": "SECRET",
        "scope": [
          "email",
          "profile",
          "openid"
        ]
      },
      "github": {
        "key": "IDTOKEN",
        "secret": "SECRET"
      },
      "discord": {
        "key": "IDTOKEN",
        "secret": "SECRET",
        "scope": [
          "identify",
          "email"
        ]
      }
    }
```
